exercises/D-logical-operators/exercise2.js

```diff
@@ -5,7 +5,47 @@
   Update the code so that you get the expected result.
 */
 
-function isNegative() {}
+function isNegative(number) {
+
+  if (number < 0){
+    return true
+  } else {
+    return false
+  }
+
+}
+
+function isBetween5and10(number) {
+
+  if ((number >= 5) && (number <= 10)) {
+    return true
+  } else {
+    return false
+  }
+
+}
+
+function isShortName(name) {
+
+  if ( name.length <= 6 ) {
+    return true
+  } else {
+    return false
+  }
+
+}
+
+function startsWithD(name) {
+
+  for ( i = 0 ; i < name.length ; i++){
+      if (name[0] == "D") {
+        return true;
+      } else {
+        return false; 
+      }
+  }
+} 
+
```

Comment:
This will work just fine but you don't need the loop; you can just check `name[0] == "D"`. The `return` statement here will mean that the loop exits after the first step - once a function hits a `return` it will stop executing

You can simplify it even more as well! `name[0] == "D"` will itself be either `true` or `false` so you can return it directly.